### PR TITLE
feat(@angular/cli): Add update cmd to update angular versions.

### PR DESCRIPTION
**Pull request by Brocco**

### CODE REVIEW

File: packages/@angular/cli/commands/update.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const Command = require('../ember-cli/lib/models/command');
+import { UpdateTask } from '../tasks/update';
+
+export interface UpdateOptions {
+  schematic?: boolean;
+}
+
+const UpdateCommand = Command.extend({
+  name: 'update',
+  description: 'Opens the official Angular API documentation for a given keyword.',
```
Comment (clydin) [inline packages/@angular/cli/commands/update.ts]: bad copy/paste?

File: packages/@angular/cli/commands/update.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  ],
+
+  anonymousOptions: [
+    '<schematic>'
+  ],
+
+  run: function(commandOptions: any, rawArgs: Array<string>) {
+    let schematic = rawArgs[0] || '@schematics/package-update:all';
+    if (!schematic.match(/:/)) {
+      schematic = `@schematics/package-update:${schematic}`;
```
Comment (clydin) [inline packages/@angular/cli/commands/update.ts]: should this be restricted to just the `package-update` collection?  maybe even validated against the list of schematics in the collection as well?  Not sure the enduser should even be aware schematics are even in use.

File: packages/@angular/cli/tasks/update.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export interface UpdateTaskOptions {
+  dryRun: boolean;
+  force: boolean;
+}
+
+export const UpdateTask: any = Task.extend({
+  run: function(schematic: string, options: UpdateTaskOptions): Promise<any> {
+    const [collectionName, schematicName] = schematic.split(':');
+
+    const schematicRunTask = new SchematicRunTask({
```
Comment (clydin) [inline packages/@angular/cli/tasks/update.ts]: Could potentially use `SchematicRunTask` directly.  Unless there's more planned.

File: packages/@angular/cli/commands/update.ts (hunk 2 above)
Comment (Brocco) [reply packages/@angular/cli/commands/update.ts]: It is not limited to just `@schematics/package-update` but if you want to use another schematic lib, you need to specify the entire schematic name (`collectionName:schematicName`). As for the validation, that error will be handled/thrown by the schematics engine.

File: packages/@angular/cli/tasks/update.ts (hunk 3 above)
Comment (Brocco) [reply packages/@angular/cli/tasks/update.ts]: Possibly, but that layer is consistent with the other command/task combinations. I'm indifferent about this.

File: packages/@angular/cli/commands/update.ts (hunk 1 above)
Comment (Brocco) [reply packages/@angular/cli/commands/update.ts]: Good catch, updating.

**Pull request merged**